tests/unittests/test_exception_handling.py

```diff
@@ -1,351 +1,290 @@
-
 from unittest import mock
 import tap_linkedin_ads.client as client
 import unittest
 import requests
 import time
+import json
+from parameterized import parameterized
+
+def get_response(status_code, json_resp={}):
+    """
+    Returns mock response
+    """
+    response = requests.Response()
+    response.status_code = status_code
+    response._content = json.dumps(json_resp).encode()
+    return response
+
+def get_400_response(*args, **kwargs):
+    """Return response for 400 error code."""
+    json_resp = {"message": "Invalid params for account.",
+                "status": 400,
+                "code": "BAD_REQUEST"}
+    return get_response(400, json_resp)
+
+def get_401_response(*args, **kwargs):
+    """Return response for 401 error code."""
+    json_resp = {"message": "The authorization has expired, please re-authorize.",
+                "status": 401,
+                "code": "UNAUTHORIZED"}
+    return get_response(401, json_resp)
+
+def get_403_response(*args, **kwargs):
+    """Return response for 403 error code."""
+    json_resp = {"message": "You do not have permission to access this resource.",
+                "status": 403,
+                "code": "FORBIDDEN"}
+    return get_response(403, json_resp)
+
+def get_404_response(*args, **kwargs):
+    """Return response for 404 error code."""
+    json_resp = {"message": "Not Found.",
+                "status": 404,
+                "code": "NOT_FOUND"}
+    return get_response(404, json_resp)
+
+def get_405_response(*args, **kwargs):
+    """Return response for 405 error code."""
+    json_resp = {"message": "The URL doesn't support this HTTP method.",
+                "status": 405,
+                "code": "METHOD_NOT_ALLOWED"}
+    return get_response(405, json_resp)
+
+def get_411_response(*args, **kwargs):
+    """Return response for 411 error code."""
+    json_resp = {"message": "Please add a defined Content-Length header.",
+                "status": 411,
+                "code": "LENGTH_REQUIRED"}
+    return get_response(411, json_resp)
+
+def get_429_response(*args, **kwargs):
+    """Return response for 429 error code."""
+    json_resp = {"message": "APT ratelimit exceeded, retry after some time.",
+                "status": 429,
+                "code": "RATELIMIT_EXCEEDED"}
+    return get_response(429, json_resp)
+
+def get_500_response(*args, **kwargs):
+    """Return response for 500 error code."""
+    json_resp = {"message": "Internal error, please retry after some time.",
+                "status": 500,
+                "code": "INTERNAL_ERROR"}
+    return get_response(500, json_resp)
+
+def get_504_response(*args, **kwargs):
+    """Return response for 504 error code."""
+    json_resp = {"message": "Gateway timed out, please retry after some time.",
+                "status": 504,
+                "code": "GATEWAY_TIMEOUT"}
```

Comment:
Can we add this specific response directly to `@parameterized.expand`?

Reply:
Updated unit tests by directly passing response or error messages in `@parameterized.expand`.